src/schedule/zephyr_dma_domain.c

```diff
@@ -0,0 +1,298 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright 2022 NXP
+//
+// Author: Paul Olaru <paul.olaru@nxp.com>
+// Author: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>
+
+#include <sof/audio/component.h>
+#include <rtos/bit.h>
+#include <rtos/interrupt.h>
+#include <rtos/timer.h>
+#include <rtos/alloc.h>
+#include <sof/lib/cpu.h>
+#include <sof/lib/dma.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/notifier.h>
+#include <sof/platform.h>
+#include <sof/schedule/ll_schedule.h>
+#include <sof/schedule/ll_schedule_domain.h>
+#include <sof/schedule/schedule.h>
+#include <sof/schedule/task.h>
+#include <ipc/topology.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+LOG_MODULE_DECLARE(ll_schedule, CONFIG_SOF_LOG_LEVEL);
+
+#if defined(__ZEPHYR__) && defined(CONFIG_IMX)
+#define interrupt_get_irq mux_interrupt_get_irq
+#define interrupt_register mux_interrupt_register
+#define interrupt_unregister mux_interrupt_unregister
+#define interrupt_enable mux_interrupt_enable
+#define interrupt_disable mux_interrupt_disable
+#endif
+
+struct driving_channel {
+	int irq; /* IRQ number */
+	struct dma_chan_data *data; /* channel data */
+	int core; /* core on which the IRQ was enabled */
+	/* pipeline task which registered the DMA IRQ */
+	struct pipeline_task *ptask;
+	bool in_use; /* true if channel is currently in use */
+	void (*work)(void *work_data); /* work to be executed by DMA IRQ */
```

Comment:
this could still be `handler`

Reply:
can we use the same terminology as zephyr LL.

Reply:
sure, will do